packages/server/graphql/mutations/helpers/removeSlackAuths.ts

```diff
@@ -0,0 +1,46 @@
+import fromTeamMemberId from 'parabol-client/utils/relay/fromTeamMemberId'
+import getRethink from '../../../database/rethinkDriver'
+
+const removeSlackAuths = async (teamMemberIds: string | string[], removeToken: boolean) => {
+  const r = await getRethink()
+  const now = new Date()
+  const teamMemberIdsArr = Array.isArray(teamMemberIds) ? teamMemberIds : [teamMemberIds]
+  const {userId} = fromTeamMemberId(teamMemberIdsArr[0])
+  const teamIds = [] as string[]
+  teamMemberIdsArr.forEach((teamMemberId) => {
+    const {teamId} = fromTeamMemberId(teamMemberId)
+    teamIds.push(teamId)
+  })
+  const existingAuths = await r
+    .table('SlackAuth')
+    .getAll(r.args(teamIds), {index: 'teamId'})
+    .filter({userId})
+    .run()
+
+  if (!existingAuths.length) {
+    const error = new Error('Auth not found')
+    return {authId: null, error}
+  }
+
+  const authIds = existingAuths.map(({id}) => id)
+  await r({
+    auth: r
+      .table('SlackAuth')
+      .getAll(r.args(authIds))
+      .update((row) => ({
+        botAccessToken: r.branch(removeToken, null, row('botAccessToken')),
+        isActive: false,
+        updatedAt: now
+      })),
+    notifications: r
+      .table('SlackNotification')
+      .getAll(r.args(teamIds), {index: 'teamId'})
+      .filter({userId})
+      .delete()
+  }).run()
+
+  const authId = authIds[0]
+  return {authId, error: null}
```

Comment:
we're skimping on the rest of the authIds!
let's return `authIds: string[]` and let the calling function decide to pick the first one.

if you really wanted to get fancy, you could use a typescript overload like:
```ts
function removeSlackAuth(teamIds: string): {authId: string}
function removeSlackAuth(teamIds: string[]): {authId: string[]}
```

but those are mainly just for showing off https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads